.github/workflows/ci.yml

```diff
@@ -21,66 +21,41 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+    - name: Set up latest uv and Python ${{ matrix.python-version }}
+      uses: astral-sh/setup-uv@v6
       with:
         python-version: ${{ matrix.python-version }}
+        activate-environment: true
     - name: Install / build
       run: |
-        python -m pip install --upgrade pip
-        pip install .
+        uv pip install .
         
   test:
-    name: test (ubuntu-latest, ${{ matrix.python-version }}, ${{ matrix.pint }} Pint, packages version ${{ matrix.version }})
+    name: test (ubuntu-latest, ${{ matrix.python-version }}, ${{ matrix.pint }} Pint, packages version ${{ matrix.resolution }})
     runs-on: "ubuntu-latest"
-    env:
-      poetry_version: 2.1.3
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.10", "3.11", "3.12", "3.13"]
         pint: ["with", "without"]
-        version: ["lock"]
-        include:
-          - python-version: "3.10"
-            version: "minimal"
-            pint: "with"
-          - python-version: "3.13"
-            version: "maximal"
-            pint: "with"
+        resolution: ["highest", "lowest-direct"]
     steps:
     - uses: actions/checkout@v4
-    - name: Install poetry
-      run: pipx install poetry==${{ env.poetry_version }}
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+    - name: Install uv and Python ${{ matrix.python-version }}
+      uses: astral-sh/setup-uv@v6
       with:
         python-version: ${{ matrix.python-version }}
-        cache: 'poetry'
+        activate-environment: true
     - name: Install udunits
       run: sudo apt-get install -y libudunits2-dev
-    - name: update lock file
-      if: matrix.version == 'maximal'
-      run: poetry update
     - name: Install dependencies
       run: |
-        poetry install --with dev
-        poetry run pip install cf_units==3.3.0
+        uv sync --resolution ${{matrix.resolution}}
```

Comment:
This is weird, it seems that uv install the extra deps. In the "without" Pint case, Pint is still installed:
https://docs.astral.sh/uv/concepts/projects/sync/#automatic-lock-and-sync

Reply:
But the doc for uv claims that it should not
https://docs.astral.sh/uv/concepts/projects/sync/#syncing-optional-dependencies